### PR TITLE
Boards page

### PR DESCRIPTION
**Pull request by malinna13**
Main route 

- Displays all created boards as a list.

- Boards are displayed with a small preview of the available information (title, description, etc.). By clicking on the element, the user redirects to the board item (Board route). There should be a button to remove the board.

- When a user tries to delete the board, he/she should receive a confirmation modal to verify if the user wants to delete the board (to avoid deleting the board by mistake). The confirmation modal should be a generic component (one for the application).

NOT DONE : Example of extra functionality (optional): global search. Search for a task by task number, name, users assigned to it, and by the text of the task description.

### CODE REVIEW

File: src/shared/components/Card/index.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ const Card: FC<ICard> = ({ children, className, ...props }) => (
     {...props}
     className={clsx(
       className,
-      'shadow-md rounded-lg bg-white overflow-hidden transition duration-300'
+      'p-6 shadow-md rounded-lg bg-white overflow-hidden transition duration-300'
```
Comment (diXrom) [inline src/shared/components/Card/index.tsx:11]: Не мутируй общий компонент, ты же можешь передать в него этот класс чрз пропc className как делала в MainPage:
`<Card className="flex justify-between cursor-pointer hover:shadow-lg">
 </Card>`

**Review by diXrom — CHANGES_REQUESTED** [review]

**Review by diXrom — COMMENTED** [review]
Добавил замечания

File: src/widgets/Header/ui/UserPanel.tsx [hunk 2]
```diff
@@ -74,7 +89,18 @@ const UserPanel = () => {
         </Transition>
       </Menu>
       <Modal isOpen={isOpen} closeModal={closeModal}>
-        <div>Добавить доску</div>
+        <h3 className="leading-loose text-lg">{t('newBoard')}</h3>
+        <form onSubmit={handleSubmit} className="flex flex-col gap-3">
```
[the remark this answers is not in the document]
Comment (malinna13) [reply src/widgets/Header/ui/UserPanel.tsx:93]: В будущем надо добавить выбор юзеров в этой форме

File: src/widgets/Header/ui/UserPanel.tsx [hunk 3]
```diff
@@ -34,6 +36,19 @@ const UserPanel = () => {
...
+  const [inputValue, setInputValue] = useState('');
+
+  function onChange(event: React.FormEvent<HTMLInputElement>) {
+    setInputValue(event.currentTarget.value);
+  }
+  const [addBoard] = useAddBoardMutation();
+
+  async function handleSubmit(event: FormEvent) {
+    event.preventDefault();
+    await addBoard({ title: inputValue, owner: user!._id, users: ['Inna'] });
```
[the remark this answers is not in the document]
Comment (malinna13) [reply src/widgets/Header/ui/UserPanel.tsx:48]: это хардкодные тестовые данные, потом доделаю выбор юзеров

File: src/widgets/Header/ui/UserPanel.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (diXrom) [reply src/widgets/Header/ui/UserPanel.tsx:93]: Понял

File: src/widgets/Header/ui/UserPanel.tsx [hunk 4]
```diff
@@ -34,6 +36,19 @@ const UserPanel = () => {
     { path: ROUTE_PATH.INDEX, icon: <FaSignOutAlt className="w-4 h-4" />, text: 'logout' },
   ];
 
+  const [inputValue, setInputValue] = useState('');
```
[the remark this answers is not in the document]
Comment (malinna13) [reply src/widgets/Header/ui/UserPanel.tsx]: вывела в отдельный компонент

File: src/pages/MainPage/index.tsx [hunk 5]
```diff
@@ -1,11 +1,65 @@
...
+      <div className="container mx-autoflex flex flex-col gap-4 ">
+        {data?.map((board) => (
+          <Link key={board._id} to={`${ROUTE_PATH.BOARDS}/${board._id}`}>
+            <Card className="flex justify-between cursor-pointer	hover:shadow-lg">
+              <h1 className="text-xl">{board.title}</h1>
+              <Button
+                className="hidden items-center gap-1 !border !border-white sm:flex"
+                onClick={() => {
+                  openModal();
+                  setBoardId(board._id);
```
[the remark this answers is not in the document]
Comment (malinna13) [reply src/pages/MainPage/index.tsx]: поменяла стейт на реф

**Review by diXrom — APPROVED** [review]

**Pull request merged**